packages/encoding/src/bech32.spec.ts

```diff
@@ -6,14 +6,69 @@ describe("Bech32", () => {
   // bech32 -e -h eth 9d4e856e572e442f0a4b2763e72d08a0e99d8ded
   const ethAddressRaw = fromHex("9d4e856e572e442f0a4b2763e72d08a0e99d8ded");
 
-  it("encodes", () => {
-    expect(Bech32.encode("eth", ethAddressRaw)).toEqual("eth1n48g2mjh9ezz7zjtya37wtgg5r5emr0drkwlgw");
+  describe("encode", () => {
+    it("works", () => {
+      expect(Bech32.encode("eth", ethAddressRaw)).toEqual("eth1n48g2mjh9ezz7zjtya37wtgg5r5emr0drkwlgw");
+    });
+
+    it("works for very short data", () => {
+      expect(() => Bech32.encode("eth", new Uint8Array(1))).not.toThrow();
+    });
+
+    it("works for very long prefixes", () => {
+      expect(() => Bech32.encode("a".repeat(81), new Uint8Array(1))).not.toThrow();
+    });
+
+    // See https://github.com/bitcoin/bips/blob/master/bip-0173.mediawiki#Bech32
+    it("works if result is 90 characters", () => {
+      const result = Bech32.encode("eth", new Uint8Array(50));
+      expect(result.length).toEqual(90);
+    });
+
+    it("throws if result exceeds 90 characters", () => {
+      expect(() => Bech32.encode("eth", new Uint8Array(51))).toThrowError(/exceeds length limit/i);
```

Comment:
Okay, a lot more tests, great 👍 But the one I was thinking about is still missing. It can be added in this block:

```ts
expect(() => Bech32.encode("p".repeat(70), new Uint8Array(20))).toThrowError(/exceeds length limit/i);
```